packages/medusa/src/strategies/batch-jobs/order/export.ts

```diff
@@ -0,0 +1,263 @@
+import { EntityManager } from "typeorm"
+import {
+  OrderDescriptor,
+  OrderExportBatchJob,
+  OrderExportBatchJobContext,
+  orderExportPropertiesDescriptors,
+} from "."
+import { AdminPostBatchesReq } from "../../../api/routes/admin/batch/create-batch-job"
+import { IFileService } from "../../../interfaces"
+import { AbstractBatchJobStrategy } from "../../../interfaces/batch-job-strategy"
+import { Order } from "../../../models"
+import { OrderService } from "../../../services"
+import BatchJobService from "../../../services/batch-job"
+import { BatchJobStatus } from "../../../types/batch-job"
+import { prepareListQuery } from "../../../utils/get-query-config"
+
+type InjectedDependencies = {
+  fileService: IFileService<any>
+  orderService: OrderService
+  batchJobService: BatchJobService
+  manager: EntityManager
+}
+
+class OrderExportStrategy extends AbstractBatchJobStrategy<OrderExportStrategy> {
+  public static identifier = "order-export-strategy"
+  public static batchType = "order-export"
+
+  public defaultMaxRetry = 3
+
+  protected readonly BATCH_SIZE = 100
+  protected readonly NEWLINE = "\r\n"
+  protected readonly DELIMITER = ";"
+
+  protected manager_: EntityManager
+  protected transactionManager_: EntityManager | undefined
+  protected readonly fileService_: IFileService<any>
+  protected readonly batchJobService_: BatchJobService
+  protected readonly orderService_: OrderService
+
+  protected readonly defaultRelations_ = ["customer", "shipping_address"]
+  protected readonly defaultFields_ = [
+    "id",
+    "display_id",
+    "status",
+    "created_at",
+    "fulfillment_status",
+    "payment_status",
+    "subtotal",
+    "shipping_total",
+    "discount_total",
+    "gift_card_total",
+    "refunded_total",
+    "tax_total",
+    "total",
+    "currency_code",
+    "region_id",
+  ]
+
+  constructor({
+    fileService,
+    batchJobService,
+    orderService,
+    manager,
+  }: InjectedDependencies) {
+    // eslint-disable-next-line prefer-rest-params
+    super(arguments[0])
+
+    this.manager_ = manager
+    this.fileService_ = fileService
+    this.batchJobService_ = batchJobService
+    this.orderService_ = orderService
+  }
+
+  async prepareBatchJobForProcessing(
+    batchJob: AdminPostBatchesReq,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    req: Express.Request
+  ): Promise<AdminPostBatchesReq> {
+    const {
+      limit,
+      offset,
+      order,
+      fields,
+      expand,
+      filterable_fields,
+      ...context
+    } = batchJob.context as OrderExportBatchJobContext
+
+    const listConfig = prepareListQuery(
+      {
+        limit,
+        offset,
+        order,
+        fields,
+        expand,
+      },
+      {
+        isList: true,
+        defaultRelations: this.defaultRelations_,
+        defaultFields: this.defaultFields_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      list_config: listConfig,
+      filterable_fields,
```

Comment:
suggestion: Should we add the stat object descriptor in a the preprocess method in case the user create a dry run job, I let you look at the product export one.

Reply:
I have added a stat object descriptor, same as product export 😄